### PR TITLE
Add ``DropRowsTransformer``

### LINKED ISSUE #2682: Add component to drop rows
In order to remove outliers / rows based on indices, we need a component that can help us do so!

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #2682

### CODE REVIEW

**Review by eccabay — APPROVED** [review]
Looks good to me, very clean!

File: evalml/pipelines/components/transformers/preprocessing/drop_rows_transformer.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+from evalml.pipelines.components.transformers import Transformer
+from evalml.utils import infer_feature_types
+
+
+class DropRowsTransformer(Transformer):
+    """Transformer to drop rows specified by row indices.
+
+
+    Arguments:
+        indices_to_drop (list): List of indices to drop in the input data.
```
Comment (bchen1116) [inline evalml/pipelines/components/transformers/preprocessing/drop_rows_transformer.py:10]: Should we include a 'Defaults to None' here?

File: evalml/tests/component_tests/test_drop_rows_transformer.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    indices_to_drop = [1, 2]
+    drop_rows_transformer = DropRowsTransformer(indices_to_drop=indices_to_drop)
+    drop_rows_transformer.fit(X, y)
+    transformed = drop_rows_transformer.transform(X, y)
+    assert_frame_equal(X_expected, transformed[0])
+    assert_series_equal(y_expected, transformed[1])
+
+    drop_rows_transformer = DropRowsTransformer(indices_to_drop=indices_to_drop)
+    fit_transformed = drop_rows_transformer.fit_transform(X)
```
Comment (bchen1116) [inline evalml/tests/component_tests/test_drop_rows_transformer.py:62]: why not pass both X and y here to `fit_transform`, since we already test for an only-X input in the test above?

File: evalml/pipelines/components/transformers/preprocessing/drop_rows_transformer.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        super().__init__(parameters=None, component_obj=None, random_seed=0)
+
+    def fit(self, X, y=None):
+        X_t = infer_feature_types(X)
+        y_t = infer_feature_types(y) if y is not None else None
+
+        if self.indices_to_drop is not None:
+            indices_to_drop_set = set(self.indices_to_drop)
+            if not indices_to_drop_set.issubset(X_t.index):
+                raise ValueError("Index does not exist in input features.")
```
Comment (bchen1116) [inline evalml/pipelines/components/transformers/preprocessing/drop_rows_transformer.py:30]: I think having a list of indices that don't exist be shown to the users would be useful here. Maybe
```python
diff = indices_to_drop_set.difference(set(X_t.index))
if len(diff):
    raise ValueError("Indices {} do not exist in input features".format(list(diff))
```
This just gives the user a more actionable message imo

File: evalml/tests/component_tests/test_drop_rows_transformer.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    assert transformed[1] is None
+
+    y = pd.Series([1, 2, 3], index=index)
+    y_expected = pd.Series([3], index=["d"])
+
+    drop_rows_transformer = DropRowsTransformer(indices_to_drop=indices_to_drop)
+    drop_rows_transformer.fit(X, y)
+    transformed = drop_rows_transformer.transform(X, y)
+    assert_frame_equal(X_expected, transformed[0])
+    assert_series_equal(y_expected, transformed[1])
```
Comment (bchen1116) [inline evalml/tests/component_tests/test_drop_rows_transformer.py:102]: Can we add tests for 1) empty list passed as `indices_to_drop` and 2) a list that has 1 index repeated (ie `[1, 2, 2]` as `indices_to_drop`?

**Review by bchen1116 — APPROVED** [review]
Looks great! Left a few comments on potential changes that would be nice, but nothing blocking!

File: evalml/tests/component_tests/test_drop_rows_transformer.py (hunk 4 above)
Comment (angela97lin) [reply evalml/tests/component_tests/test_drop_rows_transformer.py:102]: Great idea, will do 😁

File: evalml/pipelines/components/transformers/preprocessing/drop_rows_transformer.py (hunk 3 above)
Comment (angela97lin) [reply evalml/pipelines/components/transformers/preprocessing/drop_rows_transformer.py]: I like this, done!

File: evalml/tests/component_tests/test_utils.py [hunk 5]
```diff
@@ -36,21 +36,21 @@ def test_all_components(
     # Depending on the environment the detrender/Arima and/or Prophet will not be installed
 
     if has_minimal_dependencies:
-        n_components = 41
+        n_components = 42
     elif is_using_conda:
         # No prophet and no arima
-        n_components = 52
+        n_components = 53
```
Comment (angela97lin) [inline evalml/tests/component_tests/test_utils.py:42]: Typical codecov 😭
